### PR TITLE
Add admin client requirements to the process

### PR DESCRIPTION
**Pull request by darranl**
The updates I am proposing here relate to how in general the admin clients should be considered when new features are added to the application server.

The admin clients are maintained by smaller teams within the overall WildFly team, for this reason it is not going to scale if every feature requires complete admin client integration at every stability level so the general premise behind this proposal is that features can be added at Experimental or Preview level with a reduced set of admin client requirements, the admin clients can then continue to evolve to add these features and then the underlying feature can be promoted to Community / Default.

Note: This is not so much about features being added directly to the admin clients at the stability levels as in that case the admin client developer is already in control of compatibility etc...

### CODE REVIEW

File: design-doc-template.adoc [hunk 1]
```diff
@@ -97,12 +97,22 @@ categories:
 
 //== Implementation Plan
 ////
-Delete if not needed. The intent is if you have a complex feature which can 
-not be delivered all in one go to suggest the strategy. If your feature falls 
-into this category, please mention the Release Coordinators on the pull 
+Delete if not needed. The intent is if you have a complex feature which can
+not be delivered all in one go to suggest the strategy. If your feature falls
+into this category, please mention the Release Coordinators on the pull
```
Comment (bstansberry) [inline design-doc-template.adoc:102]: Please revert this change as it will conflict with other work.

File: design-doc-template.adoc [hunk 2]
```diff
@@ -97,12 +97,22 @@ categories:
...
 
+== Admin Clients
+
+////
+Identify the level of compatibility this feature will have with the existing
+admin clients (JBoss CLI and the Admin Console / HAL).
+
+Identify any follow up work that will be required in the clients and link issues
+created to track this work.
+////
```
Comment (bstansberry) [inline design-doc-template.adoc:114]: Please use the markdown style illustrated in https://github.com/wildfly/wildfly-proposals/pull/581/files#diff-1e46fd5b24b8df67bec2629d044ee017dc0e62542e582563b1abf9a29db7dfe6.

File: design-doc-template.adoc [hunk 3]
```diff
@@ -119,10 +129,10 @@ Generally a feature should have documentation as part of the PR to wildfly main,
 == Release Note Content
 ////
 Draft verbiage for up to a few sentences on the feature for inclusion in the
-Release Note blog article for the release that first includes this feature. 
+Release Note blog article for the release that first includes this feature.
```
Comment (bstansberry) [inline design-doc-template.adoc:132]: Please revert this change as it will conflict with other work.

File: design-doc-template.adoc [hunk 4]
```diff
@@ -119,10 +129,10 @@ Generally a feature should have documentation as part of the PR to wildfly main,
...
 Example article: http://wildfly.org/news/2018/08/30/WildFly14-Final-Released/.
 This content will be edited, so there is no need to make it perfect or discuss
 what release it appears in.  "See Overview" is acceptable if the overview is
-suitable. For simple features best covered as an item in a bullet-point list 
-of features containing a few words on each, use "Bullet point: <The few words>" 
+suitable. For simple features best covered as an item in a bullet-point list
+of features containing a few words on each, use "Bullet point: <The few words>"
```
Comment (bstansberry) [inline design-doc-template.adoc:137]: Please revert this change as it will conflict with other work.

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: design-doc-template.adoc (hunk 2 above)
Comment (darranl) [reply design-doc-template.adoc]: Updated

**Review by bstansberry — APPROVED** [review]
Thanks, @darranl. I'll leave merging this to you as you know what other feedback you want.

**Comment by darranl** [comment]
@hpehl / @jfdenise Can you both please have a look at this and check it makes sense for you?

This is more about how the admin clients will be updated based on other changes to the app server rather than features added directly to the admin clients.

File: FEATURE_PROCESS.adoc [hunk 5]
```diff
@@ -70,6 +70,32 @@ All hard requirements in analysis covered
 | Encouraged |Encouraged |Encouraged |Required
 //-------
 
+//-------
+
+| *Admin Clients*
+(HAL / JBoss CLI)
+
+| Management model changes must be compatible with the lowe level model
```
Comment (jfdenise) [inline FEATURE_PROCESS.adoc:78]: lowe? you mean lowest?

**Review by jfdenise — APPROVED** [review]

**Review by hpehl — APPROVED** [review]
LGTM! Thanks @darranl

File: FEATURE_PROCESS.adoc (hunk 5 above)
Comment (darranl) [reply FEATURE_PROCESS.adoc]: Thank you, I have fixed this.

**Comment by darranl** [comment]
I just sent a message to the dev list, I think will give it one more day then merge.

Being merged does not set it in stone so we can still evolve as feedback comes in / we try and work with these contraints.
